packages/nimble-components/src/rich-text/models/tests/markdown-serializer.spec.ts

```diff
@@ -1,415 +1,306 @@
-            },
-            {
-                name: 'HTML entities <&trade;>',
-                html: '&trade;',
-                markdown: '™'
-            },
-            {
-                name: 'HTML entities <&euro>',
-                html: '&euro;',
-                markdown: '€'
-            },
-            {
-                name: 'Markdown syntax strings <*>',
-                html: '*',
-                markdown: String.raw`\*`
-            },
-            {
-                name: 'Markdown syntax strings <**>',
-                html: '**',
-                markdown: String.raw`\*\*`
-            },
-            {
-                name: 'Markdown syntax strings <_>',
-                html: '_',
-                markdown: String.raw`\_`
-            }
-        ];
-
-        const focused: string[] = [];
-        const disabled: string[] = [];
-        for (const value of supportedNodesMarks) {
-            const specType = getSpecTypeByNamedList(value, focused, disabled);
-            specType(
-                `Should return ${value.name} markdown (${value.markdown}) when its respective node is passed`,
-                // eslint-disable-next-line @typescript-eslint/no-loop-func
-                () => {
-                    const node = getNode(value.html);
-
-                    expect(
-                        RichTextMarkdownSerializer.serializeDOMToMarkdown(node)
-                    ).toBe(value.markdown);
-                }
+    describe('Serialize rich text editor content to its respective markdown', () => {
+        const r = String.raw;
+
+        it('Bold', async () => {
+            await pageObject.clickFooterButton(ToolbarButton.bold);
+            await pageObject.setEditorTextContent('Bold');
+
+            expect(element.getMarkdown()).toEqual('**Bold**');
+        });
+
+        it('Italics', async () => {
+            await pageObject.clickFooterButton(ToolbarButton.italics);
+            await pageObject.setEditorTextContent('Italics');
+
+            expect(element.getMarkdown()).toEqual('*Italics*');
+        });
+
+        it('Link', async () => {
+            await pageObject.setEditorTextContent('https://nimble.ni.dev ');
+            expect(element.getMarkdown()).toEqual('<https://nimble.ni.dev> ');
+        });
+
+        it('Bold and Italics', async () => {
+            await pageObject.clickFooterButton(ToolbarButton.bold);
+            await pageObject.clickFooterButton(ToolbarButton.italics);
+            await pageObject.setEditorTextContent('Bold and Italics');
+            expect(element.getMarkdown()).toEqual('***Bold and Italics***');
+        });
+
+        it('Italics without spaces in between bold texts', async () => {
+            await pageObject.clickFooterButton(ToolbarButton.bold);
+            await pageObject.setEditorTextContent('Bold');
+            await pageObject.clickFooterButton(ToolbarButton.italics);
+            await pageObject.setEditorTextContent('italics');
```

Comment:
A suggestion to improve readability would be to rename this page object method to `appendEditorTextContent()` since `set` implies (to me) that it replaces the contents. The other method `replaceEditorTextContent()` is pretty clear already.

But I recognize this would cause churn in other test files (possibly even client tests if they're using this page object) so it could be a separate PR or something we decide isn't worthwhile.

Reply:
we will take up this work in separate PR, tracking this task [here](https://dev.azure.com/ni/DevCentral/_backlogs/backlog/ASW%20SystemLink%20LIMS/Features/?workitem=2580851)